WordPress/gutenberg
# Show review changes modal on back button in site editor

## What?

Fixes: #66248 

This PR adds an enhancement feature so that when there are unsaved changes, clicking the back button in site editor should show the Review Changes Modal instead of 'Leave Site?' alert.

## Why?


When there are unsaved changes in the site editor, clicking the back button shows generic 'Leave Site?' alert. This PR adds an enhancement feature so that when there are unsaved changes, clicking the back button should show the Review Changes Modal so that the user knows that there are changes to review/save before exiting. If there are no unsaved changes, the back button works as usual.

## How?


Add an onclick handler on the back button to open SaveView (Review Changes Modal) conditionally after checking if there are unsaved changes.

## Testing Instructions
1. Open the site editor in WordPress.
2. Make some changes to ensure there are unsaved changes.
3. Click the "Go to the Dashboard" button.
4. Verify that the "Review Changes" modal appears if there are unsaved changes. If there are no unsaved changes, verify that the editor navigates to the dashboard.


### Testing Instructions for Keyboard

1. Open the site editor in WordPress.
2. Tab to the page area and make some changes to the page content.
3. Use Tab key to go to Open Navigation button and press Enter.
4. Use the Tab key to navigate to the "Go to the Dashboard" button and press Enter to activate the button.
5. Verify that the "Review Changes" modal appears if there are unsaved changes. If there are no unsaved changes, verify that the editor navigates to the dashboard.


## Screenshots or screencast 

https://github.com/user-attachments/assets/3d5e56b0-461b-49fd-8a14-b39d959ad6ba
